v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -0,0 +1,142 @@
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+	_ extensions.PostReconcileCheckFunc,
+) (extensions.PostReconcileCheckResult, error) {
+	log.V(Debug).Info("########################## Starting Post-reconcilation for Backup Instance ##########################")
+	backupInstance, ok := obj.(*dataprotection.BackupVaultsBackupInstance)
+	if !ok {
+		return extensions.PostReconcileCheckResult{},
+			errors.Errorf("cannot run on unknown resource type %T, expected *dataprotection.BackupVaultsBackupInstance", obj)
+	}
+
+	var _ conversion.Hub = backupInstance
+
+	if owner != nil && backupInstance != nil && backupInstance.Status.Id != nil &&
+		backupInstance.Status.Properties != nil && backupInstance.Status.Properties.ProtectionStatus != nil &&
+		backupInstance.Status.Properties.ProtectionStatus.Status != nil {
+
+		protectionStatus := *backupInstance.Status.Properties.ProtectionStatus.Status
+		log.V(Debug).Info(fmt.Sprintf("########################## Protection Status is  %q ##########################", protectionStatus))
+
+		// return success for reconcilation if the state is non-retryable
+		if nonRetryableStates.Contains(protectionStatus) {
+			log.V(Debug).Info("########################## Returning PostReconcileCheckResultSuccess ##########################")
+			return extensions.PostReconcileCheckResultSuccess(), nil
+		}
+
+		// call sync api only when protection status is ProtectionError and error code is usererror
+		if protectionStatus == "ProtectionError" {
+			protectionStatusErrorCode := ""
+			protectionStatusErrorCode = strings.ToLower(*backupInstance.Status.Properties.ProtectionStatus.ErrorDetails.Code)
+			log.V(Debug).Info(fmt.Sprintf("########################## Protection Error code is  %q ##########################", protectionStatusErrorCode))
+
+			if protectionStatusErrorCode != "" && strings.Contains(protectionStatusErrorCode, "usererror") {
+				id, _ := genruntime.GetAndParseResourceID(backupInstance)
+				subscription := id.SubscriptionID
+				rg := id.ResourceGroupName
+				vaultName := id.Parent.Name
+
+				log.V(Debug).Info("########################## Starting NewBackupInstancesClient for Backup Instance ##########################")
+				clientFactory, err := armdataprotection.NewClientFactory(subscription, armClient.Creds(), armClient.ClientOptions())
+				if err != nil {
+					log.Error(err, "failed to create armdataprotection client")
+				}
+
+				var parameters armdataprotection.SyncBackupInstanceRequest
+				parameters.SyncType = to.Ptr(armdataprotection.SyncTypeDefault)
+				log.V(Debug).Info("########################## Starting BeginSyncBackupInstance for Backup Instance ##########################")
+				poller, err := clientFactory.NewBackupInstancesClient().BeginSyncBackupInstance(ctx, rg, vaultName, backupInstance.AzureName(), parameters, nil)
+				if err != nil {
+					log.Error(err, "failed BeginSyncBackupInstance to get the result")
+				}
+
+				_, err = poller.PollUntilDone(ctx, nil)
```

Comment:
How long does this take to execute? 

If this is more than a few seconds, there is a problem - while ASO is waiting on this to happen, no other reconciliations will be happening. This line is putting all reconciliation of all resources _**on hold**_ while this runs, effectively breaking ASO.

You should follow the pattern used by other long running operations - stash the URL required for the poll in an annotation on the resource, and check for it next time around.

Reply:
I need to make these changes to put the url and reuse it to check the state

Reply:
made the changes for these